tests/drivers/rtc/rtc_api/src/test_update_callback.c

```diff
@@ -18,30 +18,30 @@ static void test_rtc_update_callback_handler(const struct device *dev, void *use
 {
 	atomic_inc(&callback_called_counter);
 
-	atomic_set(&callback_test_user_data_address, (uint32_t)user_data);
+	atomic_set(&callback_test_user_data_address, (atomic_val_t)user_data);
 }
 
 ZTEST(rtc_api, test_update_callback)
 {
 	int ret;
 	uint32_t counter;
-	uint32_t address;
+	atomic_val_t address;
```

Comment:
This value is not atomic, its storing the value of an atomic. This change introduces a bug if the atomic_val_t is not a uint32_t, as you typecast it on line 54.

Reply:
The change was removed from the change set.